### PR TITLE
1133: Notebook file registration changes

### PR DESCRIPTION
**Pull request by rajmusuku**
Please review my changes to open up a notebook file from ADS

### CODE REVIEW

File: src/sql/parts/common/customInputConverter.ts [hunk 1]
```diff
@@ -48,6 +49,16 @@ export function convertEditorInput(input: EditorInput, options: IQueryEditorOpti
 			let queryPlanInput: QueryPlanInput = instantiationService.createInstance(QueryPlanInput, queryPlanXml, 'aaa', undefined);
 			return queryPlanInput;
 		}
+
+		//Notebook
+		uri = getNotebookEditorUri(input);
+		if(uri){
+			let notebookData: string = fs.readFileSync(uri.fsPath);
+			let notebookInputModel = new NotebookInputModel('modelViewId', undefined, undefined);
```
Comment (kevcunnane) [inline src/sql/parts/common/customInputConverter.ts]: 2 things here:
- For the ModelViewInput, the modelViewId is always unique. Looks like it's based on concatenating the handle with
- You need the handle to be defined or some of the code might break

File: src/sql/parts/common/customInputConverter.ts [hunk 2]
```diff
@@ -129,6 +141,29 @@ function getQueryPlanEditorUri(input: EditorInput): URI {
...
+ * If input is a supported notebook editor file (.ipynb), return it's URI. Otherwise return undefined.
+ * @param input The EditorInput to get the URI of
+ */
+function getNotebookEditorUri(input: EditorInput): URI {
+	if (!input || !input.getName()) {
+		return undefined;
+	}
+
+	// If this editor is not already of type notebook input
+	if (!(input instanceof NotebookInput)) {
```
Comment (kevcunnane) [inline src/sql/parts/common/customInputConverter.ts]: Have we tested the case where we re-convert for an existing notebook editor? What happens?

File: src/sql/parts/common/customInputConverter.ts [hunk 3]
```diff
@@ -129,6 +141,29 @@ function getQueryPlanEditorUri(input: EditorInput): URI {
 	return undefined;
 }
 
+
+/**
+ * If input is a supported notebook editor file (.ipynb), return it's URI. Otherwise return undefined.
+ * @param input The EditorInput to get the URI of
```
Comment (chlafreniere) [inline src/sql/parts/common/customInputConverter.ts]: please finish the comment

File: src/sql/parts/common/customInputConverter.ts (hunk 1 above)
Comment (AbbiePetcht) [inline src/sql/parts/common/customInputConverter.ts]: >modelViewId [](start = 52, length = 11)

This should be unique. I would use the name of the file

File: src/sql/parts/common/customInputConverter.ts [hunk 4]
```diff
@@ -48,6 +49,16 @@ export function convertEditorInput(input: EditorInput, options: IQueryEditorOpti
 			let queryPlanInput: QueryPlanInput = instantiationService.createInstance(QueryPlanInput, queryPlanXml, 'aaa', undefined);
 			return queryPlanInput;
 		}
+
+		//Notebook
+		uri = getNotebookEditorUri(input);
+		if(uri){
+			let notebookData: string = fs.readFileSync(uri.fsPath);
```
Comment (AbbiePetcht) [inline src/sql/parts/common/customInputConverter.ts]: >notebookData [](start = 7, length = 12)

Can you add the comment that we need to pass in notebook data either through notebook input or notebook service?

File: src/sql/parts/common/customInputConverter.ts (hunk 3 above)
Comment (rajmusuku) [reply src/sql/parts/common/customInputConverter.ts]: The comment says, input is the type of EditorInput object to get the URI of. It's a generic class which we are using it for Notebook scenario. We can change it in a meaningful way. I just copied it from the older method. Did not change the description as it made sense to me.

File: src/sql/parts/common/customInputConverter.ts (hunk 4 above)
Comment (rajmusuku) [reply src/sql/parts/common/customInputConverter.ts]: Done

**Pull request merged**